cs50/cs50.py

```diff
@@ -56,25 +62,33 @@ def get_int():
                     return i
             except ValueError:
                 pass
-        print("Retry: ", end="")
+
+        # temporarily here for backwards compatibility
+        if prompt is None:
+            print("Retry: ", end="")
 
 if sys.version_info.major != 3:
-    def get_long():
+    def get_long(prompt=None):
         """Read a line of text from standard input and return the equivalent long."""
         while True:
-            s = get_string();
+            s = get_string(prompt)
             if s is None:
                 return None
             if re.search(r"^[+-]?\d+$", s):
                 try:
                     return long(s, 10)
                 except ValueError:
                     pass
-            print("Retry: ", end="")
 
-def get_string():
+            # temporarily here for backwards compatibility
+            if prompt is None:
+                print("Retry: ", end="")
+
+def get_string(prompt=None):
     """Read a line of text from standard input and return it as a string."""
     try:
+        if prompt is not None:
+            print(prompt, end="")
         s = sys.stdin.readline()
         return re.sub(r"(?:\r|\r\n|\n)$", "", s)
```

Comment:
Should return `None` if not `s`, since `sys.stdin.readline` returns `""` on `EOF`?

> f.readline() returns an empty string, the end of the file has been reached

https://docs.python.org/2/tutorial/inputoutput.html#methods-of-file-objects